examples/src/guides/expression_language/runnable_history.ts

```diff
@@ -0,0 +1,90 @@
+import { ChatOpenAI } from "langchain/chat_models/openai";
+import { ChatMessageHistory } from "langchain/memory";
+import { ChatPromptTemplate } from "langchain/prompts";
+import {
+  RunnableConfig,
+  RunnableSequence,
+  RunnableWithMessageHistory,
+} from "langchain/runnables";
+import {
+  BaseListChatMessageHistory,
+  BaseMessage,
+  HumanMessage,
+} from "langchain/schema";
+import { StringOutputParser } from "langchain/schema/output_parser";
+
+// Define your session history store.
+// This is where you will store your chat history, keyed by sessionId.
+async function getListSessionHistory(): Promise<
+  (sessionId: string) => Promise<BaseListChatMessageHistory>
+> {
+  const chatHistoryStore: { [key: string]: BaseListChatMessageHistory } = {};
+
+  async function getSessionHistory(
+    sessionId: string
+  ): Promise<BaseListChatMessageHistory> {
+    if (!(sessionId in chatHistoryStore)) {
+      chatHistoryStore[sessionId] = new ChatMessageHistory();
+    }
+    return chatHistoryStore[sessionId];
+  }
+
+  return getSessionHistory;
+}
+
+// Instantiate your model and prompt.
+const model = new ChatOpenAI({});
+const prompt = ChatPromptTemplate.fromMessages([
+  ["ai", "You are a helpful assistant."],
+  ["human", "{question}"],
+]);
+
+// Create a simple runnable which passes a question to the AI.
+const runnable = RunnableSequence.from([
+  {
+    question: (messages: Array<BaseMessage>) =>
```

Comment:
Why is the input an array of messages?

Reply:
Current message + history

Reply:
This should not be necessary, the whole point of this is so that we don't have to manage history externally